feat(ast): improve `AstKind::debug_name`

DonIsaac opened the pull request:
> Part of #4445

Improve `debug_name` to show identifier names for more AST kinds.

DonIsaac commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4553?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4553** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4553?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#4552** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4552?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4551** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4551?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

rzvxa reviewed (approved):
I haven't read how every variant is named but overall this change is a good thing👍
I wanted to suggest moving display implementations to the PR down stack but I wasn't fast enough to dequeue it. I don't mind them in this PR, It is your choice if you want them in this or a new one downstack.

overlookmotel reviewed (approved):
Can't say I've read through every one of the test snapshot changes, but this seems like a good change to me.

DonIsaac committed:
feat(ast): improve `AstKind::debug_name` (#4553)

> Part of #4445

Improve `debug_name` to show identifier names for more AST kinds.
